Helpers.js

```diff
@@ -0,0 +1,242 @@
+    ui.alert('Permission denied. Aborting');
+    return true;
+  }
+}
+
+/**
+ * Show a confirmation modal indicating that we will call a 3P if the API is Green Domains
+ */
+function showConfirmationModalSustainability() {
+  var ui = SpreadsheetApp.getUi(); // Same variations.
+  var result = ui.alert(
+    'Please confirm',
+    "Carbon's usage will be done through the use of services such as CO2.js. This action will run the different models in CO2.js taking into account the results of PSI Tracker. Please run Green Domains API first otherwise the model will assume that the domains are not green hosted. Do you want to continue?",
+    ui.ButtonSet.YES_NO);
+
+  // Process the user's response.
+  if (result == ui.Button.YES) {
+    // User clicked "Yes".
+    // ui.alert('Confirmation received.'); 
+    return false;
+  } else {
+    // User clicked "No" or X in the title bar.
+    ui.alert('Permission denied. Aborting');
+    return true;
+  }
+}
+
+/**
+ * Removes triggers by handler function.
+ *
+ * Given a function name, all triggers execute said function when fired will be
+ * removed from the AppScript project.
+ *
+ * @param {string} functionName The name of the function run by the trigger.
+ */
+function deleteTriggers(functionName) {
+  for (const trigger of ScriptApp.getProjectTriggers() ?? []) {
+    if (trigger.getHandlerFunction() === functionName) {
+      ScriptApp.deleteTrigger(trigger);
+    }
+  }
+}
+
+/**
+ * Remove Batch Triggers
+ */
+function removeBatchTriggers() {
+  deleteTriggers('runBatch');
+  deleteTriggers('runBatchPSI');
+  deleteTriggers('runBatchPSIWithScreenshots');
+  deleteTriggers('runBatchCrUXHistory');
+  deleteTriggers('runBatchCrUX');
+}
+
+/**
+ * Add a toast on the bottom right corner indicating status
+ */
+function toast(title, description) {
+  SpreadsheetApp.getActive().toast(description, title, -1);
+}
+
+/**
+ * Shows a demo toast in the UI to demonstrate how the modal looks
+ */
+function toastDemo() {
+  toast("Message", "Description")
+}
+
+
+/**
+ * Dummy function to be able to click the Initialize button (UX feedback that having to double click to activate the features is suboptimal)
+ */
+function initialise() {
+  toast("Initialised", "Done")
+}
+
+/**
+ * Sets the trigger to run the tracker every day.
+ *
+ */
+function setDailyTrigger() {
+  deleteTriggers('callPSIAPI');
+  const triggerId = PropertiesService.getDocumentProperties().getProperty('triggerId');
+  if (!triggerId) {
+    const trigger = ScriptApp.newTrigger('callPSIAPI')
+      .timeBased()
+      .everyDays(1)
+      .create();
+    PropertiesService.getDocumentProperties().setProperty(
+      'triggerId', trigger.getUniqueId());
+  }
+  toast("PSI Trigger", "Done")
+}
+
+
+
+/**
+ * Sets the trigger to run the CrUX API every day.
+ *
+ */
+function setCrUXDailyTrigger() {
+  deleteTriggers('callCrUX');
+  const triggerId = PropertiesService.getDocumentProperties().getProperty('triggerId');
+  if (!triggerId) {
```

Comment:
It looks like if there was some trigger set in the past, setting the new one will not work because triggerId property is not empty. Assuming this condition here for a reason I think we need to clear triggerId property inside of deleteTriggers() function.

Alternatively, maybe we can remove if (!triggerId) condition or property in general? I am not sure I can see any other usages of this property anywhere in the code

Reply:
Real life example: client enabled CrUX trigger by mistake and now PSI trigger button is not working for them